core/unit_test/tools/TestScopedProfileRegion.cpp

```diff
@@ -0,0 +1,83 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <Kokkos_Profiling_ScopedProfileRegion.hpp>
+
+#include <gtest/gtest.h>
+
+#include <string>
+#include <stack>
+
+namespace {
+std::stack<std::string> test_region_stack;
+
+// NOTE: cannot use lambdas because they can only be converted to function
+// pointers if they don't capture anything
+void test_push_region(char const *label) {
+  ASSERT_TRUE(true) << std::string("push ") + label;
+  test_region_stack.push(label);
+}
+
+void test_pop_region() {
+  auto const &label = test_region_stack.top();
+  ASSERT_TRUE(true) << std::string("pop ") + label;
+  test_region_stack.pop();
+}
+
+}  // namespace
```

Comment:
Actually you should keep it open until the end of the file

Reply:
I looked at the `TestProfilingSection.cpp`, which does it similarly.

Reply:
We have plenty of bad examples to copy from ;)